nemo/core/neural_types/elements.py

```diff
@@ -192,4 +198,34 @@ class CategoricalValuesType(PredictionsType):
 
 
 class MaskType(PredictionsType):
-    """Element type to represent boolean mask"""
+    """Element type to represent a boolean mask"""
+
+
+class TargetType(ElementType):
+    """
+        Element type to represent a target value.
+    """
+
+
+class ClassificationTargetType(ElementType):
+    """
+        Element type to represent classification a target value, i.e. identifier of a desired class.
+    """
+
+
+class FeatureMapsType(ElementType):
+    """Element type to feature maps"""
+
+
+class ImageType(ElementType):
```

Comment:
I am wondering, shouldn't ImageType inherit from FeatureMapsType? 
Then it would make sense for conv autoencoders

Reply:
Yeah... need to think about that one... really not sure now